department-of-veterans-affairs/abd-vro
# MCP-2280: persist more claim attributes

## What was the problem?

- Need to offramp if presumptive
- Need to add more details in the persisted audit event.

Associated tickets or Slack threads:

- [MCP-2280](https://amida.atlassian.net/browse/MCP-2280)

## How does this fix it?[^1]

- check for presumptive and off-ramp claim if the test fails
- Persist all the relevant claim details in the audit_event's details field:
```
public class MasEventDetails {

  private String claimId;
  private String collectionId;
  private String veteranIcn;
  private String diagnosticCode;
  private String offRampReason;
  private String disabilityActionType;
  private Boolean presumptive;
  private boolean inScope;
  private List<String> flashIds;
  private String submissionDate;
  private String submissionSource;
}
```
## How to test this PR
- Reproduce conditions for presumptive = false. The claim is off-ramped
- Reproduce conditions for presumptive = true. The claim continues processing
- Examine the database for claim details.


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
